src/js/apps/patients/worklist/worklist_state.js

```diff
@@ -107,35 +110,66 @@ export default Backbone.Model.extend({
       [dateType]: this.formatDateRange(relativeRange, unit, dateFormat),
     };
   },
+  getDefaultStatesFilter() {
+    const isDoneOnly = this.id === 'done-last-thirty-days';
+    const { done, notDone } = this.states.groupByDone();
+
+    return isDoneOnly ? done : notDone;
+  },
+  setDefaultFilterStates() {
+    const defaultStatesFilter = this.getDefaultStatesFilter();
+    this.set({ filters: { states: defaultStatesFilter.map('id') } });
+  },
+  getSelectedStates() {
+    const defaultStatesFilterIds = this.getDefaultStatesFilter().map('id');
+    const selectedStates = this.getFilters().states;
+
+    // remove any invalid state ids (i.e. ids that don't belong to any default states)
+    return filter(selectedStates, id => contains(defaultStatesFilterIds, id)).join();
+  },
+  getFiltersCount() {
+    const filtersState = this.get('filters');
+    const defaultStatesFilter = this.getDefaultStatesFilter();
+
+    const customFiltersState = omit(filtersState, 'states');
+    const customFiltersSize = size(omit(customFiltersState, function(value) {
+      return isNull(value);
+    }));
+
+    const statesFilterSize = size(filtersState.states);
+    const defaultStatesSize = size(defaultStatesFilter.map('id'));
+
+    const statesFilterCount = statesFilterSize !== defaultStatesSize ? 1 : 0;
+
+    return customFiltersSize + statesFilterCount;
+  },
   getEntityFilter() {
-    const states = this.currentOrg.getStates();
     const filtersState = this.getFilters();
     const clinicianId = this.get('clinicianId');
     const teamId = this.get('teamId');
     const noOwner = this.get('noOwner');
-    const customFilters = omit(filtersState, 'groupId');
-    const notDoneStates = states.groupByDone().notDone.getFilterIds();
-    const doneStates = states.groupByDone().done.getFilterIds();
-
+    const customFilters = omit(filtersState, 'groupId', 'states');
+    const selectedStates = this.getSelectedStates();
     const dateFilter = this.getEntityDateFilter();
 
     const filters = {
-      'owned-by': extend({ state: notDoneStates }, dateFilter),
-      'shared-by': extend({ state: notDoneStates }, dateFilter),
+      'owned-by': dateFilter,
+      'shared-by': dateFilter,
       'new-past-day': {
         created_at: dayjs().subtract(24, 'hours').format(),
-        state: notDoneStates,
       },
       'updated-past-three-days': {
         updated_at: dayjs().startOf('day').subtract(3, 'days').format(),
-        state: notDoneStates,
       },
       'done-last-thirty-days': {
         updated_at: dayjs().startOf('day').subtract(30, 'days').format(),
-        state: doneStates,
       },
     };
 
+    if (selectedStates) {
+      filters[this.id].state = selectedStates;
+    }
```

Comment:
This makes it so `filter[state]` is either `filter[state]=id,id,id` or not added as a api request url parameter all together.

But this causes an issue when all states are un-selected.

For example:

![Screenshot 2023-01-05 at 12 22 56 PM](https://user-images.githubusercontent.com/35355575/210859019-7bed03f2-e07b-4bdb-8bb7-9c2867770675.png)

In this scenario, we want the worklist to be empty. But instead `filter[state]` is not in the url parameter, so the filter is not applied and the worklist shows all actions/flows.

Reply:
Do `filter[state]=null`?

Reply:
ah good question.  does null work?  maybe we do have to short circuit the request as you had before..  try `null` and otherwise we can confirm with BE I guess if it doesn't.

Reply:
Neither of these work:

```js
filters[this.id].state = null;
filters[this.id].state = 'null';
```

But this works:

```js
filters[this.id].state = NIL_UUID;

// result: filter[state]=00000000-0000-0000-0000-000000000000
```

Reply:
Sorry, I should have commented earlier. NIL UUID is the right solution.